src/js/Controllers/SDLController.js

```diff
@@ -22,10 +29,42 @@ class SDLController {
         }];*/
         this.externalConsentStatus = null;
         store.dispatch(setPTUWithModem(window.flags.PTUWithModemEnabled));
+
+        this.gufmCallbacks = {};
+        this.getUserFriendlyMessage = this.getUserFriendlyMessage.bind(this);
+
+        this.cfmToUpdateResult = {
+            StatusNeeded: 'UPDATE_NEEDED',
+            StatusPending: 'UPDATING',
+            StatusUpToDate: 'UP_TO_DATE'
+        }
+        this.statusMessages = {};
+        this.toastStatus = this.toastStatus.bind(this);
     }
     addListener(listener) {
         this.listener = listener
     }
+    toastStatus(status) {
+        if (!this.statusMessages[status]) {
+            this.getUserFriendlyMessage([ "StatusNeeded", "StatusPending", "StatusUpToDate" ], (response) => {
+                if (response.result.messages) {
+                    for (var msg of response.result.messages) {
+                        this.statusMessages[this.cfmToUpdateResult[msg.messageCode]] = msg;
+                    }
+                } else {
+                    this.statusMessages['UPDATE_NEEDED'] = 'Update Needed';
+                    this.statusMessages['UPDATING'] = 'Updating';
+                    this.statusMessages['UP_TO_DATE'] = 'Update Complete';
+                }
+                
+
+                this.toastStatus(status);
+            });
+        } else {
+            if (this.statusMessages[status].tts) { ttsController.queueTTS(this.statusMessages[status].tts); }
+            toast((_toast) => (<PermissionsPopup _toast={_toast} header={this.statusMessages[status].line1}/>), { duration: 2000 });
```

Comment:
```suggestion
            toast((_toast) => (<PermissionsPopup _toast={_toast} header={this.statusMessages[status].line1}/>), { duration: 5000 });
```


2000 seemed a little too fast

Reply:
They're so quick to digest, and often all three come within 2 seconds. Having all three popups covering half of my screen for 3seconds would annoy me, even knowing this won't happen often at all.

I am not opposed to extending the time a bit, but I think 5s is long, that is what I originally used.